src/tables.jl

```diff
@@ -0,0 +1,69 @@
+"""
+Constructs `Table` out of `FeatureLayer`, where every row is a `Feature` consisting of Geometry and attributes.
+```
+ArchGDAL.Table(T::Union{IFeatureLayer, FeatureLayer})
+```
+"""
+struct Table{T<:Union{IFeatureLayer, FeatureLayer}}
+    layer::T
+end
+
+getlayer(t::Table) = Base.getfield(t, :layer)
+
+function Tables.schema(layer::AbstractFeatureLayer)
+    featuredefn = layerdefn(layer)
+    fielddefns = (getfielddefn(featuredefn, i) for i in 0:nfield(layer)-1)
+    names_fields = (getname(fielddefn) for fielddefn in fielddefns)
+    types_fields = (_FIELDTYPE[gettype(fielddefn)] for fielddefn in fielddefns)
+    
+    ngeom = ArchGDAL.ngeom(featuredefn)
+    geomdefns = (ArchGDAL.getgeomdefn(featuredefn, i) for i in 0:ngeom-1)
+    geom_names = (ArchGDAL.getname(geomdefn) for geomdefn in geomdefns)
+    geom_types = (IGeometry for i in 1:ngeom)
+    
+    Tables.Schema((names_fields..., geom_names...), (types_fields..., geom_types...))
+end
+
+Tables.istable(::Type{<:Table}) = true
+Tables.rowaccess(::Type{<:Table}) = true
+Tables.rows(t::Table) = t
+
+function Base.iterate(t::Table, st = 0)
+    layer = getlayer(t)
+    st >= nfeature(layer) && return nothing
+    if iszero(st)
+        resetreading!(layer)
+    end
+    return get_row(layer), st + 1
+end
+
+function Base.getindex(t::Table, idx::Int)
+    layer = getlayer(t)
+    setnextbyindex!(layer, idx) 
+    return get_row(layer)
+end
+
+Base.IteratorSize(::Type{<:Table}) = Base.HasLength()
+Base.size(t::Table) = nfeature(getlayer(t))
+Base.length(t::Table) = size(t)
+Base.IteratorEltype(::Type{<:Table}) = Base.HasEltype()
+Base.propertynames(t::Table) = Tables.schema(getlayer(t)).names
+Base.getproperty(t::Table, s::Symbol) = [getproperty(row, s) for row in t]
+    
+function get_row(layer::IFeatureLayer)
```

Comment:
Can we name this function `nextnamedtuple(layer)` (or something to that effect to indicate the similarities with `nextfeature`) and give it a good docstring? I can see it having broader utility (beyond support for tables) for people who prefer to work with `NamedTuples` rather than `Features`.

Reply:
That sounds cool!